plugins/languages/rustaceanvim.nix

```diff
@@ -0,0 +1,271 @@
+              ];
+              max_width = null;
+              max_height = null;
+              auto_focus = false;
+            }
+          ''
+          "Options applied to floating windows. See |api-win_config|.";
+
+        crateGraph = {
+          backend = helpers.defaultNullOpts.mkStr "x11" ''
+            Backend used for displaying the graph.
+            See: https://graphviz.org/docs/outputs
+          '';
+
+          output = helpers.mkNullOrStr ''
+            Where to store the output.
+            No output if unset.
+            Relative path from `cwd`.
+          '';
+
+          full = helpers.defaultNullOpts.mkBool true ''
+            `true` for all crates.io and external crates, false only the local crates.
+          '';
+
+          enabledGraphvizBackends =
+            helpers.defaultNullOpts.mkListOf types.str
+            ''
+              [
+                "bmp" "cgimage" "canon" "dot" "gv" "xdot" "xdot1.2" "xdot1.4" "eps" "exr" "fig" "gd"
+                "gd2" "gif" "gtk" "ico" "cmap" "ismap" "imap" "cmapx" "imap_np" "cmapx_np" "jpg"
+                "jpeg" "jpe" "jp2" "json" "json0" "dot_json" "xdot_json" "pdf" "pic" "pct" "pict"
+                "plain" "plain-ext" "png" "pov" "ps" "ps2" "psd" "sgi" "svg" "svgz" "tga" "tiff"
+                "tif" "tk" "vml" "vmlz" "wbmp" "webp" "xlib" "x11"
+              ]
+            ''
+            ''
+              Override the enabled graphviz backends list, used for input validation and autocompletion.
+            '';
+
+          pipe = helpers.mkNullOrStr ''
+            Overide the pipe symbol in the shell command.
+            Useful if using a shell that is not supported by this plugin.
+          '';
+        };
+
+        openUrl = helpers.defaultNullOpts.mkLuaFn "require('rustaceanvim.os').open_url" ''
+          If set, overrides how to open URLs.
+        '';
+      };
+
+      server = {
+        autoAttach = helpers.mkNullOrStrLuaFnOr types.bool ''
+          Whether to automatically attach the LSP client.
+          Defaults to `true` if the `rust-analyzer` executable is found.
+
+          This can also be the definition of a function (`fun():boolean`).
+
+          Default:
+          ```lua
+            function()
+              local cmd = types.evaluate(RustaceanConfig.server.cmd)
+              ---@cast cmd string[]
+              local rs_bin = cmd[1]
+              return vim.fn.executable(rs_bin) == 1
+            end
+          ```
+        '';
+
+        cmd = helpers.mkNullOrStrLuaFnOr (with types; listOf str) ''
+          Command and arguments for starting rust-analyzer.
+
+          This can also be the definition of a function (`fun():string[]`).
+
+          Default:
+          ```lua
+            function()
+              return { 'rust-analyzer', '--log-file', RustaceanConfig.server.logfile }
+            end
+          ```
+        '';
+
+        settings = helpers.mkNullOrStrLuaFnOr (with types; attrsOf anything) ''
```

Comment:
We have a list of all rust-analyzer options somewhere, we should re-use that

Reply:
Well we already do it for rust-tools and the LSP options, so let's do it here too

Reply:
Ok, done